File: src/java/magma/api/list/JavaList.java
package magma.api.list;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Stream;

public record JavaList<T>(List<T> list) implements ListLike<T> {
    public JavaList() {
        this(new ArrayList<>());
    }

    @Override
    public boolean contains(final T element) {
        return list.contains(element);
    }

    @Override
    public Stream<T> stream() {
        return list.stream();
    }

    @Override
    public ListLike<T> add(final T element) {
        list.add(element);
        return this;
    }
}

File: src/windows/magma/list/JavaList.c
#include "JavaList.h"
#include "../../java/util/ArrayList.h"
#include "../../java/util/Iterator.h"
#include "../../java/util/List.h"
#include "../../java/util/stream/Stream.h"
/*

public record JavaList<Value>(List<Value> list) implements ListLike<Value> {
    public JavaList() {
        this(new ArrayList<>());
    }

    @Override
    public ListLike<Value> add(final Value element) {
        this.list.add(element);
        return this;
    }

    @Override
    public Stream<Value> stream() {
        return this.list.stream();
    }

    @Override
    public Iterator<Value> iterator() {
        return this.list.iterator();
    }
}*//*
*/